File: batches_java/batch_09_dead_code_insertion/compare.java
public class CountEven {
    private static int unusedHelper(int x, int y) {
        return x * y + 100;
    }

    private static boolean dummyCheck(int z) {
        if (z > 0) { /* no-op */ }
        return false;
    }

    public static int countEven(int[] nums) {
        int count = 0;
        for (int n : nums) {
            if (n % 2 == 0) count++;
        }
        return count;
    }

    public static void main(String[] args) {
        int[] nums = {1, 2, 3, 4, 5, 6};
        System.out.println(countEven(nums));
    }
}

File: batches_csharp/batch_09_dead_code_insertion/compare.cs
using System;

class CountEven {
    static int UnusedHelper(int x, int y) => x * y + 100;
    static bool DummyCheck(int z) { if (z > 0) { } return false; }

    static int Count(int[] nums) {
        int count = 0;
        foreach (int n in nums) if (n % 2 == 0) count++;
        return count;
    }

    static void Main() {
        int[] nums = {1, 2, 3, 4, 5, 6};
        Console.WriteLine(Count(nums));
    }
}
